custom_components/tahoma/cover.py

```diff
@@ -238,36 +250,52 @@ def icon(self):
 
     async def async_open_cover(self, **_):
         """Open the cover."""
-        await self.async_execute_command(
-            self.select_command(COMMAND_OPEN, COMMAND_UP, COMMAND_CYCLE)
-        )
+        await self.async_execute_command(self.select_command(*COMMANDS_OPEN))
 
     async def async_open_cover_tilt(self, **_):
         """Open the cover tilt."""
-        await self.async_execute_command(self.select_command(COMMAND_OPEN_SLATS))
+        await self.async_execute_command(self.select_command(*COMMANDS_OPEN_TILT))
 
     async def async_close_cover(self, **_):
         """Close the cover."""
-        await self.async_execute_command(
-            self.select_command(COMMAND_CLOSE, COMMAND_DOWN, COMMAND_CYCLE)
-        )
+        await self.async_execute_command(self.select_command(*COMMANDS_CLOSE))
 
     async def async_close_cover_tilt(self, **_):
         """Close the cover tilt."""
-        await self.async_execute_command(self.select_command(COMMAND_CLOSE_SLATS))
+        await self.async_execute_command(self.select_command(*COMMANDS_CLOSE_TILT))
 
     async def async_stop_cover(self, **_):
         """Stop the cover."""
-        await self.async_execute_command(
-            self.select_command(COMMAND_STOP, COMMAND_STOP_IDENTIFY, COMMAND_MY)
+        await self.async_cancel_or_stop_cover(
+            COMMANDS_OPEN + COMMANDS_SET_POSITION + COMMANDS_CLOSE, COMMANDS_STOP,
         )
 
     async def async_stop_cover_tilt(self, **_):
-        """Stop the cover."""
-        await self.async_execute_command(
-            self.select_command(COMMAND_STOP_IDENTIFY, COMMAND_STOP, COMMAND_MY)
+        """Stop the cover tilt."""
+        await self.async_cancel_or_stop_cover(
+            COMMANDS_OPEN_TILT + COMMANDS_SET_TILT_POSITION + COMMANDS_CLOSE_TILT,
+            COMMANDS_STOP_TILT,
         )
 
+    async def async_cancel_or_stop_cover(self, cancel_commands, stop_commands) -> None:
+        """Cancel running execution or send stop command."""
+        exec_id = next(
+            (
+                exec_id
+                for exec_id, execution in reversed(self.coordinator.executions.items())
```

Comment:
Why reverse the iterator?

Reply:
To fix the edge case where you open an directly close a cover. If you try to stop the command, it still has the execution_id from `open` and `close`. We need to cancel the last command, thus this is why I reversed the iterator.

Reply:
OK, can you add a comment please? In few months/weeks, we will forget about this.

Reply:
Added.